test/dynamo/test_misc.py

```diff
@@ -925,6 +925,314 @@ def forward(self, arg0_1: "f32[3][1]cpu", arg1_1: "f32[3][1]cpu", arg2_1: "f32[3
+            @torch._dynamo.disable
+            def foo_impl(x, y):
+                x.sin_()
+                print(y[0])
+                y[0].sin_()
+                print(y[0])
+
+            def f(x, y, z):
+                a = x[0]
+                b = z[0]
+                torch.ops.mylib.foo(a, [b, y])
+
+            orig_args = [torch.randn(2), torch.randn(2), torch.randn(2)]
+
+            [aot_eager_args, result1, graph_aot] = self.run_aot_eager(f, orig_args)
+            [inductor_args, result2, graph_inductor] = self.run_inductor(f, orig_args)
+            eager_args = pytree.tree_map_only(torch.Tensor, torch.clone, orig_args)
+            result3 = f(*eager_args)
+
+            self.assertEqual(inductor_args[2], eager_args[2])
+            self.assertEqual(inductor_args, aot_eager_args)
+
+            self.assertEqual(result3, result1)
+            self.assertEqual(result3, result2)
+
+            if torch._dynamo.config.assume_static_by_default:
+                self.assertExpectedInline(
+                    graph_aot,
+                    """\
+def forward(self, arg0_1: "f32[2][1]cpu", arg1_1: "f32[2][1]cpu", arg2_1: "f32[2][1]cpu"):
+        auto_functionalized = torch.ops.higher_order.auto_functionalized(torch.ops.mylib.foo.default, _x_base_index = 0, _x_size = (), _x_stride = (), _x_storage_offset = 0, _y_length = 2, _y_0_base_index = 1, _y_0_size = (), _y_0_stride = (), _y_0_storage_offset = 0, _y_1_base_index = 2, _y_1_size = (2,), _y_1_stride = (1,), _y_1_storage_offset = 0, _all_bases = [arg0_1, arg1_1, arg2_1])
+        getitem_1: "f32[2][1]cpu" = auto_functionalized[1]
+        getitem_2: "f32[2][1]cpu" = auto_functionalized[2]
+        getitem_3: "f32[2][1]cpu" = auto_functionalized[3];  auto_functionalized = None
+        copy_: "f32[2][1]cpu" = torch.ops.aten.copy_.default(arg0_1, getitem_1);  arg0_1 = getitem_1 = copy_ = None
+        copy__1: "f32[2][1]cpu" = torch.ops.aten.copy_.default(arg1_1, getitem_2);  arg1_1 = getitem_2 = copy__1 = None
+        copy__2: "f32[2][1]cpu" = torch.ops.aten.copy_.default(arg2_1, getitem_3);  arg2_1 = getitem_3 = copy__2 = None
+        return ()""",
+                    ignore_comments=True,
+                    ignore_empty_lines=True,
+                )
+
+            # 2. Run with inductor backend
+
+            if torch._dynamo.config.assume_static_by_default:
+                self.assertExpectedInline(
+                    graph_inductor,
+                    """\
+def forward(self, arg0_1: "f32[2][1]cpu", arg1_1: "f32[2][1]cpu", arg2_1: "f32[2][1]cpu"):
+        as_strided_default: "f32[][]cpu" = torch.ops.aten.as_strided.default(arg0_1, [], [], 0);  arg0_1 = None
+        as_strided_default_1: "f32[][]cpu" = torch.ops.aten.as_strided.default(arg1_1, [], [], 0);  arg1_1 = None
+        foo_default = torch.ops.mylib.foo.default(as_strided_default, [as_strided_default_1, arg2_1]);  as_strided_default = as_strided_default_1 = arg2_1 = foo_default = None
```

Comment:
Inductor shouldn't need to generate these as_strided calls because they're no-ops. What I suggested before (not including the size/stride/storage_offset in the case where the mutated tensor is the base) would be a forcing function to eliminate these.

Reply:
those are needed I think I already have optimization that does not generate them when we decompose:

```
                a = x[0]
                b = z[0]
 ```
 
 see the code:
 ```
     def regenerate_view(ViewInfo):
        if (
            ViewInfo.base.size() == ViewInfo.size
            and ViewInfo.base.stride() == ViewInfo.stride
            and ViewInfo.base.storage_offset() == ViewInfo.storage_offset
        ):
            return ViewInfo.base
        return torch.as_strided(
            ViewInfo.base, ViewInfo.size, ViewInfo.stride, ViewInfo.storage_offset
        )

```

Reply:
You're right, my bad.